### PR TITLE
feat: interfaceName, localized, and required added as options to pluginConfig

### PR DESCRIPTION
**Pull request by karirogg**
See payloadcms/payload#4539 and payloadcms/payload#4541

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -20,12 +20,21 @@ export type GenerateURL = <T = any>(
...
+
 export interface PluginConfig {
   collections?: string[]
   globals?: string[]
   uploadsCollection?: string
   fields?: Field[]
   tabbedUI?: boolean
+  required?: FieldOptions
+  localized?: FieldOptions
+  interfaceName?: string
```
Comment (jacobsfletch) [inline src/types.ts:37]: Problem for a different day, but we should probably pull types like this from `payload` where possible.

**Review by jacobsfletch — CHANGES_REQUESTED** [review]
Left a couple comments. Once those are addressed we can get this merged up and cut a release. Good work!

File: src/types.ts (hunk 1 above)
Comment (karirogg) [reply src/types.ts:39]: How is that done?

Comment (karirogg) [reply src/types.ts:36]: Gave it a shot - see latest commit

**Comment by JessChowdhury** [comment]
I've pulled this down to test it and made the following changes:

- Renames `fieldOptions` to `fieldOverrides` (considered going for `baseFieldOverrides`, lmk which of these feels more accurate)
- Spreads the `fieldOverrides` props instead of wiring up only required/localized
- Fixes error field styles on the custom title/desc/image fields
- Updates types for `fieldOverrides` to be more explicit
- Updates `readme` to include the new `interfaceName` and `fieldOverrides` options

Everything is working and looking good to me -  I think we are good to merge, @jacobsfletch do you want to take another look?

**Comment by jacobsfletch** [comment]
This plugin is now being maintained in the [Packages Directory](https://github.com/payloadcms/payload/tree/main/packages/plugin-seo) of the [Payload Monorepo](https://github.com/payloadcms/payload). This repo will soon be archived and all open PRs including this one will be closed. Would you be willing to re-open this same PR but within the Payload monorepo?

**Pull request closed without merging**